From 874d7c6a42e9ffa562ebb4add0474dc961419d3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Thu, 29 Oct 2020 01:13:49 +0100
Subject: [PATCH 201/417] drm/sun4i: Fix mipi-dsi panel framerate being 2/3 of
 the expected value

This happens on Pinephone and on Pinetab. Both use MIPI-DSI panels.

On Pinephone framerate is 36.6 instead of expected 55.
On Pinetab the similar 2/3 ratio holds.

Myabe there's a bug in BSP and dclk_div should really be 6 instead of 4.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/gpu/drm/sun4i/sun4i_tcon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/sun4i/sun4i_tcon.c b/drivers/gpu/drm/sun4i/sun4i_tcon.c
index b16dbf86f..fd3520076 100644
--- a/drivers/gpu/drm/sun4i/sun4i_tcon.c
+++ b/drivers/gpu/drm/sun4i/sun4i_tcon.c
@@ -398,8 +398,8 @@ static void sun4i_tcon0_mode_set_cpu(struct sun4i_tcon *tcon,
 	u32 block_space, start_delay;
 	u32 tcon_div;
 
-	tcon->dclk_min_div = SUN6I_DSI_TCON_DIV;
-	tcon->dclk_max_div = SUN6I_DSI_TCON_DIV;
+	tcon->dclk_min_div = 6;
+	tcon->dclk_max_div = 6;
 
 	sun4i_tcon0_mode_set_common(tcon, mode);
 
-- 
2.35.3

